File: android/app/src/main/java/com/nearme/app/ui/components/ButtonComponents.kt
package com.nearme.app.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.nearme.app.ui.theme.*

@Composable
fun PrimaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    isLoading: Boolean = false,
    size: ButtonSize = ButtonSize.Medium
) {
    Button(
        onClick = onClick,
        enabled = enabled && !isLoading,
        modifier = modifier.height(size.height),
        colors = ButtonDefaults.buttonColors(
            containerColor = Primary,
            contentColor = TextInverse,
            disabledContainerColor = TextTertiary,
            disabledContentColor = TextInverse
        ),
        shape = RoundedCornerShape(DesignSystem.CornerRadius.md),
        elevation = ButtonDefaults.buttonElevation(
            defaultElevation = DesignSystem.Elevation.small
        )
    ) {
        if (isLoading) {
            CircularProgressIndicator(
                modifier = Modifier.size(20.dp),
                color = TextInverse,
                strokeWidth = 2.dp
            )
        } else {
            Text(
                text = text,
                style = when (size) {
                    ButtonSize.Small -> MaterialTheme.typography.labelLarge
                    ButtonSize.Medium -> MaterialTheme.typography.bodyLarge
                    ButtonSize.Large -> MaterialTheme.typography.titleMedium
                },
                fontWeight = FontWeight.SemiBold
            )
        }
    }
}

@Composable
fun SecondaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    isLoading: Boolean = false,
    size: ButtonSize = ButtonSize.Medium
) {
    OutlinedButton(
        onClick = onClick,
        enabled = enabled && !isLoading,
        modifier = modifier.height(size.height),
        colors = ButtonDefaults.outlinedButtonColors(
            contentColor = Primary,
            disabledContentColor = TextTertiary
        ),
        border = ButtonDefaults.outlinedButtonBorder.copy(
            brush = androidx.compose.ui.graphics.SolidColor(Primary)
        ),
        shape = RoundedCornerShape(DesignSystem.CornerRadius.md),
        elevation = ButtonDefaults.outlinedButtonElevation(
            defaultElevation = DesignSystem.Elevation.small
        )
    ) {
        if (isLoading) {
            CircularProgressIndicator(
                modifier = Modifier.size(20.dp),
                color = Primary,
                strokeWidth = 2.dp
            )
        } else {
            Text(
                text = text,
                style = when (size) {
                    ButtonSize.Small -> MaterialTheme.typography.labelLarge
                    ButtonSize.Medium -> MaterialTheme.typography.bodyLarge
                    ButtonSize.Large -> MaterialTheme.typography.titleMedium
                },
                fontWeight = FontWeight.SemiBold
            )
        }
    }
}

@Composable
fun IconButton(
    icon: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    size: androidx.compose.ui.unit.Dp = DesignSystem.IconSize.xl,
    backgroundColor: Color = Surface,
    contentColor: Color = TextPrimary
) {
    Button(
        onClick = onClick,
        modifier = modifier.size(size),
        colors = ButtonDefaults.buttonColors(
            containerColor = backgroundColor,
            contentColor = contentColor
        ),
        shape = androidx.compose.foundation.shape.CircleShape,
        elevation = ButtonDefaults.buttonElevation(
            defaultElevation = DesignSystem.Elevation.small
        )
    ) {
        androidx.compose.material3.Icon(
            imageVector = androidx.compose.material.icons.Icons.Default.Add,
            contentDescription = null,
            modifier = Modifier.size(DesignSystem.IconSize.medium)
        )
    }
}

@Composable
fun FloatingActionButton(
    icon: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    size: androidx.compose.ui.unit.Dp = 56.dp,
    backgroundColor: Color = Primary
) {
    FloatingActionButton(
        onClick = onClick,
        modifier = modifier.size(size),
        containerColor = backgroundColor,
        contentColor = TextInverse,
        elevation = FloatingActionButtonDefaults.elevation(
            defaultElevation = DesignSystem.Elevation.large
        )
    ) {
        androidx.compose.material3.Icon(
            imageVector = androidx.compose.material.icons.Icons.Default.Add,
            contentDescription = null,
            modifier = Modifier.size(DesignSystem.IconSize.large)
        )
    }
}

enum class ButtonSize(val height: androidx.compose.ui.unit.Dp) {
    Small(DesignSystem.ButtonHeight.small),
    Medium(DesignSystem.ButtonHeight.medium),
    Large(DesignSystem.ButtonHeight.large)
}

@Preview(showBackground = true)
@Composable
fun ButtonComponentsPreview() {
    NearMeTheme {
        Column(
            modifier = Modifier.padding(DesignSystem.Spacing.md),
            verticalArrangement = Arrangement.spacedBy(DesignSystem.Spacing.md)
        ) {
            PrimaryButton(
                text = "Primary Button",
                onClick = {}
            )
            
            PrimaryButton(
                text = "Loading Button",
                onClick = {},
                isLoading = true
            )
            
            PrimaryButton(
                text = "Disabled Button",
                onClick = {},
                enabled = false
            )
            
            SecondaryButton(
                text = "Secondary Button",
                onClick = {}
            )
            
            Row(
                horizontalArrangement = Arrangement.spacedBy(DesignSystem.Spacing.md)
            ) {
                IconButton(
                    icon = "plus",
                    onClick = {}
                )
                
                IconButton(
                    icon = "heart",
                    onClick = {},
                    backgroundColor = Error
                )
                
                IconButton(
                    icon = "star",
                    onClick = {},
                    backgroundColor = Warning
                )
            }
            
            FloatingActionButton(
                icon = "plus",
                onClick = {}
            )
        }
    }
}

File: NearMe/Components/ButtonComponents.swift
import SwiftUI

// MARK: - Button Components
struct PrimaryButton: View {
    let title: String
    let action: () -> Void
    var isLoading: Bool = false
    var isDisabled: Bool = false
    var size: ButtonSize = .medium
    
    enum ButtonSize {
        case small, medium, large
        
        var height: CGFloat {
            switch self {
            case .small: return 36
            case .medium: return 44
            case .large: return 52
            }
        }
        
        var font: Font {
            switch self {
            case .small: return DesignSystem.Typography.buttonSmall
            case .medium: return DesignSystem.Typography.buttonMedium
            case .large: return DesignSystem.Typography.buttonLarge
            }
        }
        
        var padding: CGFloat {
            switch self {
            case .small: return DesignSystem.Spacing.sm
            case .medium: return DesignSystem.Spacing.md
            case .large: return DesignSystem.Spacing.lg
            }
        }
    }
    
    var body: some View {
        Button(action: action) {
            HStack {
                if isLoading {
                    ProgressView()
                        .progressViewStyle(CircularProgressViewStyle(tint: DesignSystem.Colors.textInverse))
                        .scaleEffect(0.8)
                } else {
                    Text(title)
                        .font(size.font)
                        .fontWeight(.semibold)
                }
            }
            .frame(height: size.height)
            .frame(maxWidth: .infinity)
            .padding(.horizontal, size.padding)
            .background(
                RoundedRectangle(cornerRadius: DesignSystem.CornerRadius.md)
                    .fill(isDisabled ? DesignSystem.Colors.textTertiary : DesignSystem.Colors.primary)
            )
            .foregroundColor(DesignSystem.Colors.textInverse)
        }
        .disabled(isDisabled || isLoading)
        .designSystemShadow(DesignSystem.Shadow.small)
    }
}

struct SecondaryButton: View {
    let title: String
    let action: () -> Void
    var isLoading: Bool = false
    var isDisabled: Bool = false
    var size: PrimaryButton.ButtonSize = .medium
    
    var body: some View {
        Button(action: action) {
            HStack {
                if isLoading {
                    ProgressView()
                        .progressViewStyle(CircularProgressViewStyle(tint: DesignSystem.Colors.primary))
                        .scaleEffect(0.8)
                } else {
                    Text(title)
                        .font(size.font)
                        .fontWeight(.semibold)
                }
            }
            .frame(height: size.height)
            .frame(maxWidth: .infinity)
            .padding(.horizontal, size.padding)
            .background(
                RoundedRectangle(cornerRadius: DesignSystem.CornerRadius.md)
                    .stroke(DesignSystem.Colors.primary, lineWidth: 2)
                    .background(
                        RoundedRectangle(cornerRadius: DesignSystem.CornerRadius.md)
                            .fill(DesignSystem.Colors.background)
                    )
            )
            .foregroundColor(isDisabled ? DesignSystem.Colors.textTertiary : DesignSystem.Colors.primary)
        }
        .disabled(isDisabled || isLoading)
        .designSystemShadow(DesignSystem.Shadow.small)
    }
}

struct IconButton: View {
    let icon: String
    let action: () -> Void
    var size: CGFloat = 44
    var backgroundColor: Color = DesignSystem.Colors.surface
    var foregroundColor: Color = DesignSystem.Colors.textPrimary
    
    var body: some View {
        Button(action: action) {
            Image(systemName: icon)
                .font(.system(size: 20, weight: .medium))
                .frame(width: size, height: size)
                .background(
                    Circle()
                        .fill(backgroundColor)
                )
                .foregroundColor(foregroundColor)
        }
        .designSystemShadow(DesignSystem.Shadow.small)
    }
}

struct FloatingActionButton: View {
    let icon: String
    let action: () -> Void
    var size: CGFloat = 56
    var backgroundColor: Color = DesignSystem.Colors.primary
    
    var body: some View {
        Button(action: action) {
            Image(systemName: icon)
                .font(.system(size: 24, weight: .semibold))
                .frame(width: size, height: size)
                .background(
                    Circle()
                        .fill(backgroundColor)
                )
                .foregroundColor(DesignSystem.Colors.textInverse)
        }
        .designSystemShadow(DesignSystem.Shadow.large)
    }
}

// MARK: - Button Previews
struct ButtonComponents_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: DesignSystem.Spacing.md) {
            PrimaryButton(title: "Primary Button", action: {})
            
            PrimaryButton(title: "Loading Button", action: {}, isLoading: true)
            
            PrimaryButton(title: "Disabled Button", action: {}, isDisabled: true)
            
            SecondaryButton(title: "Secondary Button", action: {})
            
            HStack(spacing: DesignSystem.Spacing.md) {
                IconButton(icon: "plus", action: {})
                IconButton(icon: "heart", action: {}, backgroundColor: DesignSystem.Colors.error)
                IconButton(icon: "star", action: {}, backgroundColor: DesignSystem.Colors.warning)
            }
            
            FloatingActionButton(icon: "plus", action: {})
        }
        .padding()
        .background(DesignSystem.Colors.background)
    }
}
